### PR TITLE
Feature incompressible

### PR DESCRIPTION
**Pull request by talbring**
We have separated the incompressible direct and adjoint solver from the compressible ones. This will ease further improvements of the incomp. implementation. Furthermore, we removed the free surface solver, as we cannot maintain it anymore.

There are still some things related to the compr. formulation in the __inc._ files. For example heat related stuff or the computation of the local time step is still based on the Prandtl numbers. This will be removed soon.

However, I wanted to have this already in develop because merging is really a pain ....

Along with this separation, I also fixed all compiler warnings that were reported in travis.

### CODE REVIEW

**Comment by talbring** [comment]
Some additional comments:

- There are a lot of duplicated routines now, because they look the same for compressible/incompressible. These will be eventually moved to the CSolver. We have also plans for modifing the MPI routines in the solvers to make them more generic.

- I cannot make sure that some features that are currently not tested in Travis still work. For example things like Harmonic Balance for incompressible flows (in case it was working before). We should find out whats working and whats not so that we can maybe throw appropriate error messages.

- It might have happened that I accidentally did something wrong during the merge. Maybe I reverted some cosmetic changes that do not affect the solution ... If you see something, let me know. So I request **ALL** of you to have a look at the solver structure. But keep in mind that just looking at the diff might be confusing as it is not showing all changes correctly.

**Comment by economon** [comment]
Thanks for submitting, Tim. This is an important change to the structure, so yes, it's best to get it in asap. I will start going through it myself soon.

I would add that this is likely to cause some conflicts for folks, so please look for specific issues related to your own branches that will cause problems.

As for the capabilities of the incompressible solver, you can assume many of them from the compressible solver were not actually available/working. Things like harmonic balance, grid motion, rotating frame, etc. need to be implemented properly in the new standalone solver for the incompressible version. This was one of the main motivations for moving things around (and so we can keep expanding). The only capabilities we should expect are those in the regressions, i.e., basic Euler/N-S/RANS primals and adjoints.

**Comment by rsanfer** [comment]
@economon No ALE terms, right?

**Comment by talbring** [comment]
ALE is one of the things missing right now. But this is something we want to target asap, to bring the incompressible solver on the same level as the compressible one. Furthermore we noticed that there is a lot of potential for further optimizations in terms of memory and speed. Conjugate Heat transfer (compressible and incompressible) is also on the way :)

PS: I will fix the conflicts soon.

**Comment by talbring** [comment]
Feedback (positive or negative) from everybody is requested :) I get conflicts with every new pull request at the moment and it costs me always some time to resolve them ...

**Comment by LaSerpe** [comment]
Hi Tim, this sounds great, I agree we should get this merged as soon as possible.
I made some extra test and everything looks good to me.  So far I didn't see any conflict with what I'm working on right now. 
Lets wait for some other review but on my side I'd say we can go ahead with this PR.

ciao!

File: SU2_CFD/src/variable_adjoint_mean_inc.cpp [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	IntBoundary_Jump = new su2double [nVar];
+	for (iVar = 0; iVar < nVar; iVar++)
+		IntBoundary_Jump[iVar] = 0.0;
+  
+  /*--- Allocate space for the time spectral source terms ---*/
+  if (config->GetUnsteady_Simulation() == HARMONIC_BALANCE) {
+		TS_Source = new su2double[nVar];
+		for (iVar = 0; iVar < nVar; iVar++)
+			TS_Source[iVar] = 0.0;
+	}
```
Comment (arubino) [inline SU2_CFD/src/variable_adjoint_mean_inc.cpp]: There are not anymore TS source terms but now just HB sources. Can you please check this?

File: SU2_CFD/src/variable_adjoint_mean_inc.cpp [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  
+  bool dual_time = ((config->GetUnsteady_Simulation() == DT_STEPPING_1ST) ||
+                    (config->GetUnsteady_Simulation() == DT_STEPPING_2ND));
+  
+  /*--- Array initialization ---*/
+	Psi = NULL;
+	ForceProj_Vector = NULL;
+	ObjFuncSource = NULL;
+	IntBoundary_Jump = NULL;
+	TS_Source = NULL;
```
Comment (arubino) [inline SU2_CFD/src/variable_adjoint_mean_inc.cpp]: Again here there is TS_Source.

File: SU2_CFD/src/variable_direct_mean_inc.cpp [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+    }
+	}
+  
+	/*--- Allocate space for the time spectral source terms ---*/
+  
+  if (config->GetUnsteady_Simulation() == HARMONIC_BALANCE) {
+		TS_Source = new su2double[nVar];
+		for (iVar = 0; iVar < nVar; iVar++) TS_Source[iVar] = 0.0;
+	}
+
```
Comment (arubino) [inline SU2_CFD/src/variable_direct_mean_inc.cpp]: Still the old names for TS_Source instead of the new HB_Source.

Comment (arubino) [inline SU2_CFD/src/variable_direct_mean_inc.cpp]: Also the comment should be /*--- Allocate space for harmonic balance source terms

File: SU2_CFD/include/variable_structure.inl [hunk 4]
```diff
@@ -858,6 +783,122 @@ inline void CAdjNSVariable::SetVelSolutionOldDVector(void) { for (unsigned short
...
+inline void CAdjIncEulerVariable::SetObjFuncSource(su2double *val_ObjFuncSource) { for (unsigned short iVar = 0; iVar < nVar; iVar++) ObjFuncSource[iVar] = val_ObjFuncSource[iVar]; }
+
+inline void CAdjIncEulerVariable::SetIntBoundary_Jump(su2double *val_IntBoundary_Jump) { for (unsigned short iVar = 0; iVar < nVar; iVar++) IntBoundary_Jump[iVar] = val_IntBoundary_Jump[iVar]; }
+
+inline void CAdjIncEulerVariable::SetPhi_Old(su2double *val_phi) { for (unsigned short iDim = 0; iDim < nDim; iDim++) Solution_Old[iDim+1]=val_phi[iDim]; };
+
+inline void CAdjIncEulerVariable::SetTimeSpectral_Source(unsigned short val_var, su2double val_source) { TS_Source[val_var] = val_source; }
+
+inline su2double CAdjIncEulerVariable::GetTimeSpectral_Source(unsigned short val_var) { return TS_Source[val_var]; }
+
```
Comment (arubino) [inline SU2_CFD/include/variable_structure.inl]: These methods are now changed to HB only for example.

Comment (arubino) [inline SU2_CFD/include/variable_structure.inl]: One additional note is that I am not sure HB incompressible would work like it is now, we can however create a testcase and if needed do the proper modifications.

**Comment by economon** [comment]
Working on this next. Just fyi...

**Pull request merged**